test/utils_spec_runner.py

```diff
@@ -229,7 +229,19 @@ def check_result(self, expected_result, result):
 
             return True
         else:
-            self.assertEqual(result, expected_result)
+
+            def _helper(expected_result, result):
+                if isinstance(expected_result, abc.Mapping):
+                    for i in expected_result.keys():
+                        _helper(expected_result[i], result[i])
```

Comment:
I believe this needs to be `self.assertEqual` since only the top level documents should ignore missing keys.